openssl/src/ssl/test/mod.rs

```diff
@@ -309,17 +309,19 @@ fn pending() {
 
 #[test]
 fn state() {
+    const EXPECTED_STATE_STRING_LONG: &str = "SSL negotiation finished successfully";
+
     let server = Server::builder().build();
 
     let s = server.client().connect();
-    #[cfg(not(boringssl))]
+    #[cfg(not(any(boringssl, awslc)))]
     assert_eq!(s.ssl().state_string().trim(), "SSLOK");
     #[cfg(boringssl)]
     assert_eq!(s.ssl().state_string(), "!!!!!!");
-    assert_eq!(
-        s.ssl().state_string_long(),
-        "SSL negotiation finished successfully"
-    );
+    #[cfg(awslc)]
+    assert_eq!(s.ssl().state_string(), EXPECTED_STATE_STRING_LONG);
+
+    assert_eq!(s.ssl().state_string_long(), EXPECTED_STATE_STRING_LONG);
```

Comment:
What's going on here?

Reply:
AWS-LC delegates `state_string()` down to `state_string_long()` so the output is 1-1 the same. Where as BoringSSL does not do that and just output the fixed string that you were testing for.